---a/net/minecraft/world/demo/DemoWorldServer.java
+++b/net/minecraft/world/demo/DemoWorldServer.java
@@ -8,17 +8,24 @@
 import net.minecraft.world.storage.ISaveHandler;
 import net.minecraft.world.storage.WorldInfo;
 
-public class DemoWorldServer extends WorldServer
-{
-    private static final long demoWorldSeed = (long)"North Carolina".hashCode();
+import org.bukkit.World.Environment;
+import org.bukkit.generator.ChunkGenerator;
+
+public class DemoWorldServer extends WorldServer {
+
+    private static final long demoWorldSeed = (long) "North Carolina".hashCode();
     public static final WorldSettings demoWorldSettings = (new WorldSettings(demoWorldSeed, WorldSettings.GameType.SURVIVAL, true, false, WorldType.DEFAULT)).enableBonusChest();
     private static final String __OBFID = "CL_00001428";
 
-    public DemoWorldServer(MinecraftServer server, ISaveHandler saveHandlerIn, WorldInfo worldInfoIn, int dimensionId, Profiler profilerIn)
-    {
-        super(server, saveHandlerIn, worldInfoIn, dimensionId, profilerIn);
+    // SpigotMCP Start -- Add enviroment and chunk generator
+    public DemoWorldServer(MinecraftServer server, ISaveHandler saveHandlerIn, WorldInfo worldInfoIn, int dimensionId, Profiler profilerIn, Environment environment, ChunkGenerator generator) {
+        super(server, saveHandlerIn, worldInfoIn, dimensionId, profilerIn, environment, generator);
+        // SpigotMCP End
         this.worldInfo.populateFromWorldSettings(demoWorldSettings);
     }
-}
+}
